airflow/providers/google/cloud/operators/cloud_build.py

```diff
@@ -15,186 +15,678 @@
+from airflow.utils.decorators import apply_defaults
 
-    * It is required to provide the source and only one type can be given,
-    * It is possible to provide the source as the URL address instead dict.
 
-    :param body: The request body.
-        See: https://cloud.google.com/cloud-build/docs/api/reference/rest/Shared.Types/Build
-    :type body: dict
+class CloudBuildCancelBuildOperator(BaseOperator):
+    """
+    Cancels a build in progress.
+
+    :param id_: The ID of the build.
+    :type id_: str
+    :param project_id: Optional, Google Cloud Project project_id where the function belongs.
+        If set to None or missing, the default project_id from the GCP connection is used.
+    :type project_id: Optional[str]
+    :param retry: Optional, a retry object used  to retry requests. If `None` is specified, requests
+        will not be retried.
+    :type retry: Optional[Retry]
+    :param timeout: Optional, the amount of time, in seconds, to wait for the request to complete.
+        Note that if `retry` is specified, the timeout applies to each individual attempt.
+    :type timeout: Optional[float]
+    :param metadata: Optional, additional metadata that is provided to the method.
+    :type metadata: Optional[Sequence[Tuple[str, str]]]
+    :param gcp_conn_id: Optional, the connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: Optional[str]
+
+    :rtype: dict
     """
-    def __init__(self, body: Dict) -> None:
-        self.body = deepcopy(body)
-
-    def _verify_source(self):
-        is_storage = "storageSource" in self.body["source"]
-        is_repo = "repoSource" in self.body["source"]
-
-        sources_count = sum([is_storage, is_repo])
 
-        if sources_count != 1:
-            raise AirflowException(
-                "The source could not be determined. Please choose one data source from: "
-                "storageSource and repoSource."
-            )
+    template_fields = ("project_id", "id_", "gcp_conn_id")
 
-    def _reformat_source(self):
-        self._reformat_repo_source()
-        self._reformat_storage_source()
+    @apply_defaults
+    def __init__(
+        self,
+        id_: str,
+        project_id: Optional[str] = None,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.id_ = id_
+        self.project_id = project_id
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
 
-    def _reformat_repo_source(self):
-        if "repoSource" not in self.body["source"]:
-            return
+    def execute(self, context):
+        hook = CloudBuildHook(gcp_conn_id=self.gcp_conn_id)
+        result = hook.cancel_build(
```

Comment:
Previously, this operator used BuildProcessor to slightly modify input for security reasons and to make it more user-friendly. Can you restore these features?